Old production tool real test

VictorBarbosaMartins opened the pull request:
Closes #741.

After getting the old production tool (`simtools-production`) to life, now it is time to test it properly. In this PR, I made some changes in the tool to allow it running simtel_array for specific input CORSIKA files. Before, only a rather non-trivial structure of sub-directories was allowed, which made difficult to test the tool with `--array_only` option. 

Major changes:
- It is now possible to run e.g. `simtools-production --productionconfig ./tests/resources/prod_config_test.yml --task simulate --array_only --submit_command local --output_path . --corsika_files tests/resources/tel_output_10GeV-2-gamma-20deg-CTAO-South.zst` and check the output in `ls simtel-data/North/proton/data/run000001_proton_za020deg_azm000deg_North_TestLayout_test-production.simtel.zst`
- This new feature is tested by checking whether the output file exists (expanded a bit the test_applications_from_config). This is good and we should use for all the applications which generate output files and are not ecsv.

Remaining point, which I deem for unimportant is whether we want or not to change the name of the test files to include the zero padding. We have a couple of files in resources still in the old format, e.g. `tests/resources/run1_proton_za20deg_azm0deg_North_1LST_test-lst-array.corsika.zst`.

orelgueta commented on tests/unit_tests/corsika/test_corsika_runner.py:
```diff
@@ -43,7 +43,7 @@ def corsika_runner(corsika_config_data, io_handler, simtel_path, db_config):
 def corsika_file(io_handler):
     corsika_file = io_handler.get_input_data_file(
-        file_name="run1_proton_za20deg_azm0deg_North_1LST_test-lst-array.corsika.zst", test=True
+        file_name="run1_proton_za20deg_azm0deg_north_1lst_test-lst-array.corsika.zst", test=True
```
Please do not change the name of the file, it is produced with the specific naming and capitalisation and should not change unless there is good reason for it. In this case it seems to be a bug that changes the name of the file, not an actual good reason.

orelgueta commented on tests/unit_tests/simtel/test_simtel_runner_array.py:
```diff
@@ -49,7 +49,7 @@ def simtel_runner(array_model, simtel_path):
 def corsika_file(io_handler):
     corsika_file = io_handler.get_input_data_file(
-        file_name="run1_proton_za20deg_azm0deg_North_1LST_test-lst-array.corsika.zst", test=True
+        file_name="run1_proton_za20deg_azm0deg_north_1lst_test-lst-array.corsika.zst", test=True
```
Same comment as above:
> Please do not change the name of the file, it is produced with the specific naming and capitalisation and should not change unless there is good reason for it. In this case it seems to be a bug that changes the name of the file, not an actual good reason.

orelgueta reviewed (changes requested):
A few minor issues and the file name point which points to a bug somewhere.

VictorBarbosaMartins commented on simtools/applications/production.py:
```diff
@@ -270,7 +286,24 @@ def main():
+                if not Path(corsika_file).exists():
+                    msg = (
+                        f"CORSIKA file {corsika_file} does not exist. Please run the "
+                        f"production with the `--showers_only` option first or point the "
```
There is one variable: `{corsika_file}` maybe you didn't see it because it is an iterator

VictorBarbosaMartins commented:
Hi Orel. Thanks for the comments. I addressed your comments and returned the name to what it was before. I had not seen before the `.lower()` in the argument of the new parameter, I think that might solve the problem I was having before.
